iOSVideoApp/iOSVideoApp/views/CameraViewController.swift

```diff
@@ -0,0 +1,146 @@
+            
+            //create input, verify it, add it.
+            guard let cameraInput = try? AVCaptureDeviceInput(device: camera),
+                captureSession.canAddInput(cameraInput) else {
+                    fatalError("Error adding camera to capture session")
+            }
+            captureSession.addInput(cameraInput)
+            
+            //Additional configuration/Session Preset.
+            if captureSession.canSetSessionPreset(.hd1920x1080) {
+                captureSession.sessionPreset = .hd1920x1080
+            }
+            
+            //Microphone input
+            
+            
+            //Add outputs
+            guard captureSession.canAddOutput(fileOutput) else {
+                fatalError("Cannot save video: Output issue.")
+            }
+            captureSession.addOutput(fileOutput)
+            
+            
+            captureSession.commitConfiguration()
+            cameraView.session = captureSession
+            //Start/Stop session
+        }
+        
+        private func bestCamera() -> AVCaptureDevice {
+            //checking the camera types.
+            if let ultraWideCamera = AVCaptureDevice.default(.builtInUltraWideCamera, for: .video, position: .back) {
+                return ultraWideCamera
+            }
+            if let wideAngleCamera = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: .back) {
+                return wideAngleCamera
+            }
+            fatalError("No camera available - are you on a simulator?")
+        }
+        
+        //TODO: Fix this shit yo to pop us to the view controller
+    
+        @IBAction func recordButtonPressed(_ sender: Any) {
+            toggleRecord()
+        }
+        
+        private func toggleRecord() {
+            if fileOutput.isRecording {
+                fileOutput.stopRecording()
+            } else {
+                fileOutput.startRecording(to: newRecordingURL(), recordingDelegate: self)
+            }
+        }
+        
+        /// Creates a new file URL in the documents directory
+        private func newRecordingURL() -> URL {
+            let documentsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+            
+            let formatter = ISO8601DateFormatter()
+            formatter.formatOptions = [.withInternetDateTime]
+            
+            let name = formatter.string(from: Date())
+            let fileURL = documentsDirectory.appendingPathComponent(name).appendingPathExtension("mov")
+            return fileURL
+        }
+        
+        private func playMovie(url: URL) {
+            player = AVPlayer(url: url)
+            let playerView = VideoPlayerView()
+            //this should be in viewdidload so it is only added once and add it to the sub view only once (see FIXME below)
+            playerView.player = player
+            // top left corner (Fullscreen, you'd need a close button)
```

Comment:
Or a swipe gesture, or something like that

Reply:
Can you explain? Had issues with this video player.